### ISSUE TITLE
"Cannot edit in read-only editor" tooltip may not show in web version

### ISSUE THREAD

**Issue by bwateratmsft** [microsoft/vscode#113854]
- VSCode Version: 1.53.0-insider (in web browser)
- OS Version: Win10 20H2

Steps to Reproduce:

1. Open a VSCode window in-browser, e.g. a Codespace
1. Open a read only file
1. Try typing text into the file

Expected:
Tooltip that says "Cannot edit in read-only editor":
![image](https://user-images.githubusercontent.com/36966225/103701707-786ad500-4f74-11eb-95c0-fc25b38f6b96.png)

Actual:
No tooltip

NOTE: This issue does not occur in the desktop version of VSCode, only web. It also does not occur when trying to delete text--i.e. hit Backspace--but does occur when trying to add text.

**Comment by jrieken** [comment]
I can reproduce with Safari

**Comment by alexdima** [comment]
Yes, on the web the `<textarea>` of a read only editor is using markup which sets the attribute `readonly` here -- https://github.com/microsoft/vscode/blob/676bb6b100e2bb4aecca791c8cde737c74126f32/src/vs/editor/browser/controller/textAreaHandler.ts#L136 . 

This initially came in as a way to stop mobile devices to show the pop-up keyboard via https://github.com/microsoft/vscode/commit/0040158a0d438bf9c1584edc9733bf8708e1360c so maybe we could restrict this to only web + mobile, but we don't have a good `isMobile` condition at this point.

**Comment by yume-chan** [comment]
Maybe the [inputmode="none"](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode) attribute  can be used to prevent showing the virtual keyboard. I have tested it on Chrome for Android and Safari for iOS.

**Comment by alexdima** [comment]
🤖 Thanks for reporting this. This was fixed in https://github.com/microsoft/vscode/commit/02b394a1a5c43ebc830cc6e56162d4fea68e4751 which introduced the `domReadOnly` option to decouple the HTML `readonly` attribute from the editor's read-only state. The tooltip now appears correctly on web. Shipped in VS Code 1.55.0. Closing as resolved — please open a new issue if you still encounter this on current builds.
